web/main.ts

```diff
@@ -2591,6 +2637,22 @@ class GitGraphView {
 				this.changeFileViewType(GG.FileViewType.List);
 			});
 
+			document.getElementById('cdvParentToggle')?.addEventListener('click', () => {
+				const expandedCommit = this.expandedCommit!;
+				const parentIndex = expandedCommit.parentIndex;
+				const parents = expandedCommit.commitDetails!.parents;
+
+				let nextParentIndex = parentIndex < parents.length ? parentIndex + 1 : 1;
+				while(!loadedParents.includes(parents[nextParentIndex - 1])) {
+					if(nextParentIndex === parentIndex) {

```

Comment:
@mhutchie What do you think about this bit of logic? It is meant to prevent infinite loops that shouldn't really be possible.

Reply:
You can completely remove the concept of the `loadedParents`, as you should be able to work with just the `expandedCommit.commitDetails.parents` (we want the user to be able to view the changes with any parent of the commit, not just the ones that are loaded).